src/Compilers/Core/Portable/SynthesizedLocalKind.cs

```diff
@@ -232,6 +232,12 @@ internal enum SynthesizedLocalKind
         /// TODO: Avoid using lambdas and display classes for implementation of relaxation stubs and remove this kind.
         /// </summary>
         DelegateRelaxationReceiver = 0x101,
+
+        /// <summary>
+        /// The builder type of an interpolated string, when using the interpolated string builder pattern.
+        /// This synthesized local is allowed to have a val escape
+        /// </summary>
+        InterpolatedStringBuilder = 0x102,
```

Comment:
>InterpolatedStringBuilder [](start = 8, length = 25)

I performed a search for "InterpolatedStringBuilder.SynthesizedLocalKind" in files in this PR. CodeFlow shows only one match. It appears no one is specifically checking for this value. It is quite possible that the textual search doesn't find all references for this member, but, if it does, it doesn't look like there is a real reason behind adding this member (at least in this PR).

Reply:
After the refactor, I am using it.

---
In reply to: [610658103](https://github.com/dotnet/roslyn/pull/52422#discussion_r610658103) [](ancestors = 610658103)

Reply:
After the full refactor and addressing your comments, I'm not using this value directly, so I _could_ turn it back into a local rewriting temp. However, I'd prefer to keep it as a distinct value, for my ease of debugging.

---
In reply to: [612030082](https://github.com/dotnet/roslyn/pull/52422#discussion_r612030082) [](ancestors = 612030082,610658103)